dask/bag/core.py

```diff
@@ -1710,6 +1701,90 @@ def build_bag_kwargs(n):
     return return_type(dsk, name, npartitions)
 
 
+def map_partitions(func, *args, **kwargs):
+    """Apply a function to every partition across one or more bags.
+
+    Note that all ``Bag`` arguments must be partitioned identically.
+
+    Parameters
+    ----------
+    func : callable
+    *args, **kwargs : Bag, Item, Delayed, or object
+        Arguments and keyword arguments to pass to ``func``.
+
+    Examples
+    --------
+    >>> import dask.bag as db
+    >>> b = db.from_sequence(range(1, 101), npartitions=10)
+    >>> def div(nums, den=1):
+    ...     return [num / den for num in nums]
+
+    Using a python object:
+
+    >>> hi = b.max().compute()
+    >>> hi
+    100
+    >>> b.map_partitions(div, den=hi).take(5)
+    (0.01, 0.02, 0.03, 0.04, 0.05)
+
+    Using an ``Item``:
+
+    >>> b.map_partitions(div, den=b.max()).take(5)
+    (0.01, 0.02, 0.03, 0.04, 0.05)
+
+    Note that while both versions give the same output, the second forms a
+    single graph, and then computes everything at once, and in some cases
+    may be more efficient.
+    """
+    name = 'map-partitions-%s-%s' % (funcname(func),
+                                     tokenize(func, args, kwargs))
+
+    # Extract bag arguments, build initial graph
+    bags = []
+    dsk = {}
+    for vals in [args, kwargs.values()]:
+        for a in vals:
+            if isinstance(a, (Bag, Item, Delayed)):
+                dsk.update(a.dask)
```

Comment:
a.dask -> ensure_dict(a.dask)

Sharedicts are somewhat expensive to turn into dicts.  This only matters for large graphs.